tests/protocol/test_hostplugin.py

```diff
@@ -862,6 +862,125 @@ def test_should_report(self):
         self.assertEqual(False, actual)
 
 
+class TestHostPluginVmSettings(HttpRequestPredicates, AgentTestCase):
```

Comment:
Move from test_wire. The implementation is almost identical to the original, but instead of testing via  WireClient.update_goal_state() now we test via HostPluginProtocol.fetch_vm_settings()